src/config/field-overrides/dataSet.js

```diff
@@ -1,7 +1,98 @@
+import React from 'react';
 import OrganisationUnitTreeMultiSelect from '../../forms/form-fields/orgunit-tree-multi-select';
 import DataSetElementField from './data-set/DataSetElementField.component';
 import DataInputPeriods from './data-set/DataInputPeriods.component';
 import PeriodTypeDropDown from '../../forms/form-fields/period-type-drop-down';
+import Checkbox from '../../forms/form-fields/check-box';
+import {RadioButton, RadioButtonGroup} from "material-ui/RadioButton";
+import log from "loglevel";
+
+class CustomTestComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            renderAsTabs: props.value,
+            displayOptions: this.parseDisplayOptions()
+        };
+
+    }
+
+    parseDisplayOptions = () =>  {
+        try {
+            return this.props
+                && this.props.model['displayOptions']
+                && JSON.parse(this.props.model['displayOptions'])
+        } catch (e) {
+            log.error(e);
+            return undefined
+        }
+    }
+
+    updateTabsDirection = (tabsDirection) => {
+        const newDisplayOptions = {
+            ...this.state.displayOptions,
+            tabsDirection
+        }
+        this.setState((prevState, _) => ({
+            ...prevState,
+            displayOptions: newDisplayOptions
+        }));
+        this.props.model.displayOptions = JSON.stringify(newDisplayOptions)
+    }
+
+    onDisplayOptionsChanged = (event) =>  {
+        const tabsDirection = event.target.value
+        this.updateTabsDirection(tabsDirection)
+
+    }
+
+    onDisplayAsTabsChanged = (event) =>  {
+        const renderAsTabs = event.target.value
+        const tabsDirection =
+            renderAsTabs
+                ? 'horizontal'
+                : undefined
+
+        this.setState((prevState) => ({
+            ...prevState,
+            renderAsTabs
+        }));
+        this.props.model.renderAsTabs = renderAsTabs
+        this.updateTabsDirection(tabsDirection)
+    }
+
+
+    render() {
+        const state = this.state;
+        return <div>
+           <Checkbox
+            labelText="Render as tabs"
```

Comment:
Should translate this (and other labels). You can add the d2 context to get access to i18n by wrapping the component with addD2Context:

```
import addD2Context from 'd2-ui/lib/component-helpers/addD2Context';

// ...
   ['renderAsTabs', {
        component: addD2Context(CustomTestComponent)
    }]

// in component:
// this.context.d2.i18n.getTranslation('render_as_tabs')
```

Add keys in `i18n_module_en.properties`.

Reply:
YES. I completely forgot about translations. Thanks!